modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/DefaultDslContext.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.config.dsl;
+
+import org.mule.runtime.extension.api.ExtensionManager;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.xml.dsl.api.resolver.DslResolvingContext;
+
+import java.util.Optional;
+
+/**
+ * Default implementation of {@link DslResolvingContext} that uses
+ * the {@link ExtensionManager} to provide the required {@link ExtensionModel}s
+ *
+ * @since 4.0
+ */
+class DefaultDslContext implements DslResolvingContext {
```

Comment:
just to clarify, this is the one that will be used when an application is being deployed on a real mule right?